cypress/e2e/models/migration/applicationinventory/assessment.ts

```diff
@@ -192,28 +192,51 @@ export class Assessment extends Application {
         super.edit(updatedValues);
     }
 
+    retake_questionnaire(
+        risk,
+        stakeholders?: Array<string>,
+        stakeholderGroups?: Array<string>
+    ): void {
+        let applicationOpen: boolean;
+        Application.open();
+        this.selectApplication();
+        clickItemInKebabMenu(this.name, "Assess");
+        cy.wait(SEC);
+        clickByText(button, "Retake");
+        // This check can't be done from the test because the alert message is displayed for a short time
+        // and we are in the midddle of answering a questionnaire at this point.
+        checkSuccessAlert(
+            commonView.alertTitle,
+            `Success alert:Success! Assessment discarded for ${this.name}.`
+        );
+        this.perform_assessment(risk, stakeholders, stakeholderGroups, (applicationOpen = false));
+    }
+
     take_questionnaire(): void {
         clickByText(button, "Take");
     }
 
     perform_assessment(
         risk,
         stakeholders?: Array<string>,
-        stakeholderGroups?: Array<string>
+        stakeholderGroups?: Array<string>,
+        applicationOpen = true
     ): void {
         if (stakeholders == undefined && stakeholderGroups == undefined) {
             expect(
                 false,
                 "At least one arg out of stakeholder or stakeholder groups must be provided !"
             ).to.equal(true);
         } else {
-            Application.open();
-            selectItemsPerPage(100);
-            this.selectApplication();
-            clickItemInKebabMenu(this.name, "Assess");
-            cy.wait(6000);
-            this.take_questionnaire();
-            cy.wait(SEC);
+            // These steps are not required for an assessment retake.
+            if (applicationOpen) {
+                Application.open();
+                this.selectApplication();
+                clickItemInKebabMenu(this.name, "Assess");
```

Comment:
Suggestion : These three lines can go in a method called clickAssessButton()

and that can be called in two methods .But it is an optional change .